### PR TITLE
Require three dashes (---) at the beginning of YAML pipelines

### PR DESCRIPTION
**Pull request by mtojek**
Closes https://github.com/elastic/package-spec/issues/53

### CODE REVIEW

**Comment by mtojek** [comment]
@ycombinator and I discussed it offline and agreed to reach out first to the Elasticsearch team for explanation for non-YAML spec behavior (three dashes shouldn't be mandatory here).

**Comment by ycombinator** [comment]
> @ycombinator and I discussed it offline and agreed to reach out first to the Elasticsearch team for explanation for non-YAML spec behavior (three dashes shouldn't be mandatory here).  

Related Elasticsearch issue: https://github.com/elastic/elasticsearch/issues/63247

File: versions/1/data_stream/spec.yml [hunk 1]
```diff
@@ -39,14 +39,10 @@ spec:
...
         contents:
-        - description: Default ingest pipeline definition
-          type: file
-          pattern: '^default\.(yml|json)$'
-          required: true
         - description: Supporting ingest pipeline definitions in YAML
           type: file
           pattern: '.+\.yml'
-          contentMediaType: "application/x-yaml"
+          contentMediaType: "application/x-yaml; require-document-dashes=true"
```
Comment (ycombinator) [inline versions/1/data_stream/spec.yml:45]: Let's add a TODO or similar comment here referencing https://github.com/elastic/elasticsearch/issues/63247 so we know to revisit this change if the ES issue is resolved on their end.

File: code/go/internal/validator/folder_item_spec.go [hunk 2]
```diff
@@ -105,8 +107,21 @@ func loadItemContent(itemPath, mediaType string) ([]byte, error) {
...
+	}
+
+	basicMediaType, params, err := mime.ParseMediaType(mediaType)
+	if err != nil {
+		return nil, errors.Wrapf(err, "invalid media type (%s)", mediaType)
+	}
+
+	switch basicMediaType {
 	case "application/x-yaml":
+		if v, _ := params["require-document-dashes"]; v == "true" && !bytes.HasPrefix(itemData, []byte("---\n")) {
```
Comment (ycombinator) [inline code/go/internal/validator/folder_item_spec.go:121]: Let's add a TODO or similar comment here referencing https://github.com/elastic/elasticsearch/issues/63247 so we know to revisit this change if the ES issue is resolved on their end.

File: versions/1/data_stream/spec.yml (hunk 1 above)
Comment (mtojek) [reply versions/1/data_stream/spec.yml:46]: Added!

File: code/go/internal/validator/folder_item_spec.go (hunk 2 above)
Comment (mtojek) [reply code/go/internal/validator/folder_item_spec.go:122]: Added!

Comment (ycombinator) [reply code/go/internal/validator/folder_item_spec.go:122]: Did you mean to reference https://github.com/elastic/package-spec/pull/54 in your comments (as opposed to https://github.com/elastic/elasticsearch/issues/63247)?

Comment (mtojek) [reply code/go/internal/validator/folder_item_spec.go:122]: We had an offline discussion to leave the reference to https://github.com/elastic/package-spec/pull/54 .

**Review by ycombinator — APPROVED** [review]
LGTM.

**Pull request merged**